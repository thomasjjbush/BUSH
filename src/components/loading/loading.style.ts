import styled from 'styled-components';

export const StyledLoading = styled.main`
    align-items: center;
    display: flex;
    flex: 1;
    flex-direction: column;
    justify-content: center;
`;
