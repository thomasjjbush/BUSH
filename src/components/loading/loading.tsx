import React, { FunctionComponent, ReactElement } from 'react';
import { useTheme } from 'styled-components';
import { Image } from '../';
import { StyledLoading } from './loading.style';
import { Theme } from '../../types';

export const Loading: FunctionComponent = (): ReactElement => {
    const { images } = useTheme() as Theme;

    return (
        <StyledLoading data-test-id="loading">
            <h1>Loading</h1>
            <Image src={images.loading} size={200} width={50} />
        </StyledLoading>
    );
};
