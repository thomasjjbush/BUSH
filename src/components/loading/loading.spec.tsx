import React from 'react';
import { shallow } from 'enzyme';
import { Loading } from './loading';

describe('Thumbnail', () => {
    it('should match snapshot', () => {
        expect(shallow(<Loading />)).toMatchSnapshot();
    });
});
