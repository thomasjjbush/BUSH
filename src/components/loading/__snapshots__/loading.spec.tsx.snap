// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Thumbnail should match snapshot 1`] = `
<styled.main
  data-test-id="loading"
  styles="
    align-items: center;
    display: flex;
    flex: 1;
    flex-direction: column;
    justify-content: center;
"
>
  <h1>
    Loading
  </h1>
  <Image
    size={200}
    src="theme.images.loading"
    width={50}
  />
</styled.main>
`;
